src/test/java/org/tarantool/jdbc/JdbcExceptionHandlingTest.java

```diff
@@ -0,0 +1,322 @@
+package org.tarantool.jdbc;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+import org.junit.jupiter.api.function.ThrowingConsumer;
+import org.tarantool.CommunicationException;
+import org.tarantool.TarantoolConnection;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.sql.DatabaseMetaData;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Properties;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.tarantool.jdbc.SQLDatabaseMetadata.FORMAT_IDX;
+import static org.tarantool.jdbc.SQLDatabaseMetadata.INDEX_FORMAT_IDX;
+import static org.tarantool.jdbc.SQLDatabaseMetadata.SPACE_ID_IDX;
+import static org.tarantool.jdbc.SQLDatabaseMetadata.SPACES_MAX;
+import static org.tarantool.jdbc.SQLDatabaseMetadata._VINDEX;
+import static org.tarantool.jdbc.SQLDatabaseMetadata._VSPACE;
+import static org.tarantool.jdbc.SQLDriver.PROP_SOCKET_TIMEOUT;
+
+import org.tarantool.TarantoolControl;
+
+public class JdbcExceptionHandlingTest {
+    protected static TarantoolControl control;
+
+    /**
+     * We cannot mock TarantoolConnection constructor, so need listening
+     * tarantool instance to prevent a test failure.
+     */
+    @BeforeAll
+    public static void setupEnv() throws Exception {
+        control = new TarantoolControl();
+        control.start("jdk-testing");
+        control.waitStarted("jdk-testing");
+    }
+
+    @AfterAll
+    public static void teardownEnv() throws Exception {
+        control.stop("jdk-testing");
+        control.waitStopped("jdk-testing");
+    }
```

Comment:
But we can mock the socket. I have proposed a fix in db072ed24df4076122447a3a6fdb4f8852b67af4

Reply:
Thanks! I pushed the fix on top of the existing commits.

Rebase it into the first version of JdbcExceptionHandlingTest will need changes from the following commits and I think it is okay to fix it afterwards, because the goal to have tests working on each commit is achieved anyway.